src/Topology.tsx

```diff
@@ -609,24 +609,39 @@ export class Topology extends React.Component<Props, {}> {
 
         // dispatch node per groups
         node.children.forEach(child => {
-            var field = child.wrapped.data.Type
-            if (!field) {
+            var gid = node.id
+
+            var nodeType = child.wrapped.data.Type
+            if (!nodeType) {
                 return
             }
-
-            var weight = child.wrapped.getWeight()
+            gid += "_" + nodeType
 
             // group only nodes being at the same level, meaning weight
-            var gid = node.id + "_" + field + "_" + weight
-
-            var name = field + '(s)'
+            var weight = child.wrapped.getWeight()
+            gid += "_" + weight
+ 
+            var name = nodeType + "(s)"
+            if (nodeType == "pod") {
```

Comment:
I would prefer to not have something specific to node metadata in this file, did you try playing with the configuration file ? is there anything missing in it ?
https://github.com/skydive-project/skydive-ui/blob/master/assets/config.js#L104

Reply:
thanks, I will see how I can move specifics outside of main flow and into config.js

Reply:
so for now I have implemented two helper functions within Topology:
- groupName()
- groupGid()

the question is what value do we get by moving it from Topology.tsx (which is the correct context for this logic) to config.js ... more so this means we move from the safer TypeScript to JavaScript. So I am really not sure of the value of doing so as its' obvious this is not mere configuration.

Reply:
so let's call this logic or data transformer if config is not good. I think it is sane to decouple any specific logic from the rendering engine otherwise we will add there a lot of stuff depending on specific use cases. Here that is k8s but later that will be NSM and calico, block storage, etc. And if someone wants to tweak the view that will be doable just overriding the "config".

Obviously this can be converted to typescript

Reply:
ok ... would you suggest groupName() and groupGid() be moved to the config.js ?

Reply:
In fact I started to work on this few time ago but did not finish, so there is already a field for `groupGid` which is `groupBy`. There are few references in the config file and in the topology code. 
I would use only one function returning the `key` or `gid` and the `name`.